### PR TITLE
feat: Added links internalization

### PR DESCRIPTION
**Pull request by nirikash**
Ref.: #DEV-11660

### CODE REVIEW

File: src/main/java/ch/sbb/polarion/extension/pdf/exporter/util/html/HtmlLinksHelper.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+                new ExternalCssInternalizer(fileResourceProvider)
+        ));
+    }
+
+    public HtmlLinksHelper(Set<LinkInternalizer> linkInliners) {
+        this.linkInliners = linkInliners;
+    }
+
+    public String internalizeLinks(String htmlContent) {
+        boolean enablingProperty = PdfExporterExtensionConfiguration.getInstance().getInternalizeExternalCss();
```
Comment (yurtsevich-sbb) [inline src/main/java/ch/sbb/polarion/extension/pdf/exporter/util/html/HtmlLinksHelper.java:30]: Isn't name like `linksInternalizationEnabled` better than `enablingProperty`?

File: src/main/java/ch/sbb/polarion/extension/pdf/exporter/util/HtmlProcessor.java [hunk 2]
```diff
@@ -1137,6 +1140,10 @@ public String replaceImagesAsBase64Encoded(String html) {
         return html;
     }
 
+    public String internalizeLinks(String html) {
```
Comment (grigoriev) [inline src/main/java/ch/sbb/polarion/extension/pdf/exporter/util/HtmlProcessor.java:1143]: why do you need this method?

Comment (nirikash) [reply src/main/java/ch/sbb/polarion/extension/pdf/exporter/util/HtmlProcessor.java:1143]: The HtmlToPdfConverter and PdfConverter don't call HtmlLinksHelper directly, but only through the HtmlProcessor.internalizeLinks().
The reason is minimizing injected dependencies in HtmlToPdfConverter and PdfConverter and logically operation is html processing like replacing base64 encoded images (just too large to be implemented directly in HtmlProcessor)

File: src/main/java/ch/sbb/polarion/extension/pdf/exporter/util/html/HtmlLinksHelper.java (hunk 1 above)
Comment (nirikash) [reply src/main/java/ch/sbb/polarion/extension/pdf/exporter/util/html/HtmlLinksHelper.java:30]: I thought the context is already defined by class name and method name, and wouldn't like repeat it. But if it more readable, ok to rename

**Review by grigoriev — APPROVED** [review]

**Pull request merged**